desc/equilibrium/coords.py

```diff
@@ -436,71 +477,52 @@ def _map_clebsch_coordinates(
         Stopping tolerance.
     maxiter : int
         Maximum number of Newton iterations.
-    full_output : bool, optional
-        If True, also return a tuple where the first element is the residual from
-        the root finding and the second is the number of iterations.
     kwargs : dict, optional
         Additional keyword arguments to pass to ``root_scalar`` such as ``maxiter_ls``,
         ``alpha``.
 
     Returns
     -------
-    out : ndarray
-        Shape (k, 3).
-        DESC computational coordinates [ρ, θ, ζ].
-    info : tuple
-        2 element tuple containing residuals and number of iterations for each point.
-        Only returned if ``full_output`` is True.
+    theta : ndarray
+        Shape (num iota, num alpha, num zeta).
+        DESC computational coordinates θ at given input meshgrid.
 
     """
     # noqa: D202
 
     # Root finding for θₖ such that r(θₖ) = αₖ(ρ, θₖ, ζ) − α = 0.
-    def rootfun(theta, alpha, rho, zeta, iota):
-        nodes = jnp.array([rho.squeeze(), theta.squeeze(), zeta.squeeze()], ndmin=2)
-        A = L_basis.evaluate(nodes)
-        lmbda = A @ L_lmn
-        alpha_k = theta + lmbda - iota * zeta
-        return _fixup_residual(alpha_k - alpha, period).squeeze()
+    def rootfun(theta, alpha, c_m):
+        c = (jnp.exp(1j * modes * theta) * c_m).real.sum()
+        alpha_k = theta + c
+        return _fixup_residual(alpha_k - alpha, period)
 
-    def jacfun(theta, alpha, rho, zeta, iota):
-        # Valid everywhere except θ such that θ+λ = k period where k ∈ ℤ.
-        nodes = jnp.array([rho.squeeze(), theta.squeeze(), zeta.squeeze()], ndmin=2)
-        A1 = L_basis.evaluate(nodes, (0, 1, 0))
-        lmbda_t = jnp.dot(A1, L_lmn)
-        return 1 + lmbda_t.squeeze()
+    def jacfun(theta, alpha, c_m):
+        # Valid everywhere except where θ+λ−ιω = k period where k ∈ ℤ.
+        dc_dt = ((1j * jnp.exp(1j * modes * theta) * c_m).real * modes).sum()
+        return 1 + dc_dt
 
     def fixup(x, *args):
         return _periodic(x, period)
 
-    vecroot = jit(
-        vmap(
-            lambda x0, *p: root_scalar(
-                rootfun,
-                x0,
-                jac=jacfun,
-                args=p,
-                fixup=fixup,
-                tol=tol,
-                maxiter=maxiter,
-                full_output=full_output,
-                **kwargs,
-            )
+    @partial(jnp.vectorize, signature="(),(),(m)->()")
+    def vecroot(theta0, alpha, c_m):
```

Comment:
I think you'll still have the issue @YigitElma mentioned about recompilation here. The fix would be to make rootfun/jacfun/vecroot global (private) functions rather than define them locally within this function

Reply:
In `ku/nufft` I jitted the outer function as well.